modules/exploits/windows/local/nscp_pe.rb

```diff
@@ -0,0 +1,239 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpClient
+  include ::Msf::Exploit::Powershell
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NSClient++ 0.5.2.35 - Privilege escalation',
+        'Description' => %q{
+          This module allows an attacker with an unprivileged windows account to gain admin access on windows system and start a shell.
+          For this module to work, both web interface of NSClient++ and `ExternalScripts` feature should be enabled.
+          You must also know where is the NSClient config file as it is used to read the admin password which is stored in clear text.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [               # This module is kind of mix of the two following POCs :
+            'kindredsec', # POC on www.exploit-db.com
+            'BZYO',       # POC on www.exploit-db.com
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '48360'],
+            ['EDB', '46802']
+          ],
+        'Platform' => %w[windows],
+        'Arch' => [ARCH_X64],
+        'Targets' =>
+          [
+            [
+              'Windows',
+              {
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :windows_powershell
+              }
+            ]
+          ],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-10-20',
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          },
+        'DefaultOptions' => { 'SSL' => true, 'RHOSTS' => 'DOESNT_MATTER', 'RPORT' => 8443 }
+      )
+    )
+
+    register_options [
```

Comment:
```suggestion
        'DefaultOptions' => { 'SSL' => true, 'RPORT' => 8443 }
      )
    )
    
    deregister_options('RHOSTS')
    register_options [
```

Reply:
I did that before but then I couldn't use `send_request_cgi` anymore... Do you know how to send local http request with a session (as I don't want to send that request on my attacker machine but on the targeted one of course)

Reply:
I tested this myself and wasn't able to recreate a bug with `send_request_cgi` with the `RHOSTS` deregistered, what error msg were you getting?

Reply:
Hello ! I'm getting this error : 
![image](https://user-images.githubusercontent.com/84333864/121895755-df10cd80-cd20-11eb-8c99-85d0c8d2e794.png)
I think `send_request_cgi` tries to concatenate `RPORT` with `RHOST` but as the latter doesn't exist, it crashes.